frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,384 @@
+  const [selectedNodeData, setSelectedNodeData] = React.useState([]);
+
+  const onSelect = (
+    event: React.MouseEvent<HTMLButtonElement>,
+    isSelected: boolean,
+    virtualRowIndex: number,
+    data: any[],
+  ) => {
+    event.preventDefault();
+    let newSelectedRowData = _.cloneDeep(selectedNodeData);
+
+    // clone `data` in case previous Firehose updates added elements, preserve existing selection state
+    // use findIndex to findout the exact node selected, 59 episode, use set to keep selected nodes
+    _.each(data, (row, index: number) => {
+      if (index < newSelectedRowData.length) {
+        // preserve existing selection state
+        newSelectedRowData[index] = {
+          ...row,
+          uid: row.metadata.uid,
+          selected: newSelectedRowData[index].selected,
+        };
+      } else {
+        // set initial selection state from storage here if necessary...for now, initialize it false
+        newSelectedRowData.push({ ...row, uid: row.metadata.uid, selected: false });
+      }
+    });
+
+    if (virtualRowIndex !== -1) {
+      // set the selection based on virtualRowIndex node, it should exist in the array now
+      newSelectedRowData[virtualRowIndex].selected = isSelected;
+    } else {
+      // selectAll
+      newSelectedRowData = _.map(newSelectedRowData, (row) => ({ ...row, selected: isSelected }));
+    }
+    setSelectedNodeData(newSelectedRowData);
+
+    setSelectedNodesCnt(_.filter(newSelectedRowData, 'selected').length);
+  };
+
+  React.useEffect(() => {
+    selectedNodesCnt >= minSelectedNode ? setCreateBtnDisabled(false) : setCreateBtnDisabled(true);
+  }, [selectedNodesCnt]);
+
+  const handleRequestSizeInputChange = (obj) => {
+    setRequestSizeUnit(obj.unit);
+    setRequestSizeValue(obj.value);
+  };
+
+  const handleStorageClass = (sc) => {
+    setStorageClass(_.get(sc, 'metadata.name'));
+  };
+
+  const OCSNodeList = (nodeProps) => (
+    <NodeList
+      {...nodeProps}
+      data={nodeProps.data}
+      customData={selectedNodeData}
+      onSelect={(event, isSelected, virtualRowIndex) =>
+        onSelect(event, isSelected, virtualRowIndex, nodeProps.data)
+      }
+    />
+  );
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setProgress(true);
+    setError('');
+
+    const labelPath = '/metadata/labels';
+    const promises = [];
+
+    // labeling the selected nodes
+    const selectedData = _.filter(selectedNodeData, 'selected');
+    promises.push(
+      ...selectedData.map(function(node) {
+        const labels = SelectorInput.arrayify(_.get(node, labelPath.split('/').slice(1)));
+        const patch = [
+          {
+            op: labels.length ? 'replace' : 'add',
+            value: SelectorInput.objectify(labels),
+            path: labelPath,
+          },
+        ];
+        patch[0].value = { ...patch[0].value, 'cluster.ocs.openshift.io/openshift-storage': '' };
```

Comment:
can we make `'cluster.ocs.openshift.io/openshift-storage': ''` as `const nodeLabel`?